src/native/http_request_response.c

```diff
@@ -534,25 +550,27 @@ JNIEXPORT void JNICALL Java_software_amazon_awssdk_crt_http_HttpStreamBase_httpS
 
     /* global ref this because now the callbacks will be firing, and they will release their reference when the
      * stream callback sequence completes. */
-    cb_data->java_http_stream_base = (*env)->NewGlobalRef(env, j_http_stream_base);
-    aws_atomic_store_int(&cb_data->activated, 1);
+    binding->java_http_stream_base = (*env)->NewGlobalRef(env, j_http_stream_base);
     if (aws_http_stream_activate(stream)) {
-        aws_atomic_store_int(&cb_data->activated, 0);
-        (*env)->DeleteGlobalRef(env, cb_data->java_http_stream_base);
+        (*env)->DeleteGlobalRef(env, binding->java_http_stream_base);
         aws_jni_throw_runtime_exception(
             env, "HttpStream activate failed with error %s\n", aws_error_str(aws_last_error()));
+    } else {
+        /* Succeed activating the stream, holding the binding to make sure it lives across all the callbacks until
+         * stream completes */
+        aws_http_stream_binding_acquire(binding);
     }
```

Comment:
we should only +1 the refcount if the stream is changing from not-activated -> activated

Nothing is stopping java users from calling `stream.activate()` multiple times.

except I don't think there's currently any way to detect whether or not the stream is activated based on the public C API. `aws_http_stream_activate()` will return AWS_OP_SUCCESS if it's called multiple times. We thought that would nice for people :|

Reply:
Good Catch!

Options:
- Add an activated check function in C, I believe the python binding has the exact same issue, it can use that function as well. [here](https://github.com/awslabs/aws-crt-python/blob/main/source/http_stream.c#L315)
- From the language level to skip calling activate in native code multiple times, which may benefit the performance?? 

Prefer the first one as we will need to handle the case for all the language bindings and it's easily to miss, I think

Reply:
Option 3: shutdown_complete callback in C, and the bindings can use that to know when to cleanup